### PR TITLE
Add py-nanoplot and py-nanostat

### PR DESCRIPTION
**Pull request by Pandapip1**
Untested - please validate!

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-kaleido/package.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    """Static image export for web-based visualization libraries with zero dependencies"""
+
+    homepage = "https://github.com/wdecoster/nanostat"
+    pypi = "kaleido/kaleido-0.2.1.post1-py2.py3-none-manylinux2014_armv7l.whl"
+
+    maintainers("Pandapip1")
+
+    version("0.2.1", sha256="d313940896c24447fc12c74f60d46ea826195fc991f58569a6e73864d53e5c20")
+
+    depends_on("python@:3.4")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/py-kaleido/package.py]: We don't support this version of Python anymore:
```
py-kaleido: dependency on python@:3.4 cannot be satisfied by known versions of python
    happening in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/py-kaleido/package.py
```

**Review by alalazo — DISMISSED** [review]

File: var/spack/repos/builtin/packages/py-kaleido/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-kaleido/package.py]: Where do you see that Python 3.4 and older is required? If you're referring to:
```python
install_requires=[
        "pathlib ; python_version<'3.4'",
    ],
```
this translates to "pathlib is required when using Python 3.4 and older", not "Python 3.4 and older is required".

Comment (Pandapip1) [reply var/spack/repos/builtin/packages/py-kaleido/package.py]: Oh, got it.

File: var/spack/repos/builtin/packages/py-kaleido/package.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+class PyKaleido(PythonPackage):
+    """Static image export for web-based visualization libraries with zero dependencies"""
+
+    homepage = "https://github.com/wdecoster/nanostat"
+    pypi = "kaleido/kaleido-0.2.1.post1-py2.py3-none-manylinux2014_armv7l.whl"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-kaleido/package.py]: This wheel only works on Linux ARM. Can we download and build from GitHub source instead?

File: var/spack/repos/builtin/packages/py-kaleido/package.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+    maintainers("Pandapip1")
+
+    version("0.2.1", sha256="d313940896c24447fc12c74f60d46ea826195fc991f58569a6e73864d53e5c20")
+
+    depends_on("python")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-kaleido/package.py]: You can remove this line, it is implied and added by PythonPackage

File: var/spack/repos/builtin/packages/py-nanoget/package.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    homepage = "https://github.com/wdecoster/nanoget"
+    pypi = "nanoget/nanoget-1.19.3.tar.gz"
+
+    maintainers("Pandapip1")
+
+    version("1.19.3", sha256="da981810edb1cbe42cbbfbe5fcf753f29bf5555204cd51256b28a284a036a71b")
+
+    depends_on("python@3")
+    depends_on("py-biopython")
+    depends_on("py-pysam@0.10.0.0:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nanoget/package.py]:
```suggestion
    depends_on("py-pysam@0.10.0.1:")
```

File: var/spack/repos/builtin/packages/py-nanoget/package.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    """Functions to extract information from Oxford Nanopore sequencing data and alignments"""
+
+    homepage = "https://github.com/wdecoster/nanoget"
+    pypi = "nanoget/nanoget-1.19.3.tar.gz"
+
+    maintainers("Pandapip1")
+
+    version("1.19.3", sha256="da981810edb1cbe42cbbfbe5fcf753f29bf5555204cd51256b28a284a036a71b")
+
+    depends_on("python@3")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nanoget/package.py]: You can remove this line, Spack only supports Python 3 anyway

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nanoget/package.py]: Missing build dep on setuptools

File: var/spack/repos/builtin/packages/py-nanoget/package.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    depends_on("py-biopython")
+    depends_on("py-pysam@0.10.0.0:")
+    depends_on("py-pandas@2.0.0:")
+    depends_on("py-numpy")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nanoget/package.py]: All deps other than setuptools should be `type=("build", "run")`

File: var/spack/repos/builtin/packages/py-nanomath/package.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    """A few simple math function for other Oxford Nanopore processing scripts"""
+
+    homepage = "https://github.com/wdecoster/nanomath"
+    pypi = "nanomath/nanomath-1.3.0.tar.gz"
+
+    maintainers("Pandapip1")
+
+    version("1.3.0", sha256="c35a024b10b34dd8f539cefed1fd69e0a46d18037ca48bed63c7941c67ae028e")
+
+    depends_on("python@3")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nanomath/package.py]: Remove

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nanomath/package.py]: Missing build dep on setuptools

File: var/spack/repos/builtin/packages/py-nanomath/package.py [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    depends_on("py-pandas")
+    depends_on("py-numpy@1.8:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nanomath/package.py]:
```suggestion
    depends_on("py-numpy@1.9:")
```

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nanomath/package.py]: Missing build/run dep on py-python-deprecated

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nanomath/package.py]: Should all be type=build/run

File: var/spack/repos/builtin/packages/py-nanoplot/package.py [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    """Plotting scripts for long read sequencing data"""
+
+    homepage = "https://github.com/wdecoster/NanoPlot"
+    pypi = "NanoPlot/NanoPlot-1.42.0.tar.gz"
+
+    maintainers("Pandapip1")
+
+    version("1.42.0", sha256="0f8fd2cffd33a346b3306716058c6cb4091c931e8ab502f10b17a28749e8b6d9")
+
+    depends_on("python@3")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nanoplot/package.py]: Remove

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nanoplot/package.py]: Build dep on setuptools

File: var/spack/repos/builtin/packages/py-nanoplot/package.py [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+    depends_on("py-biopython")
+    depends_on("py-pysam@0.10.0.0:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nanoplot/package.py]:
```suggestion
    depends_on("py-pysam@0.10.0.1:")
```

File: var/spack/repos/builtin/packages/py-nanoplot/package.py [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    depends_on("py-pandas@1.1.0:")
+    depends_on("py-numpy@1.16.5:")
+    depends_on("py-scipy")
+    depends_on("py-python-dateutil")
+    depends_on("py-nanoget@1.19.1:")
+    depends_on("py-nanomath@1.0.0:")
+    depends_on("py-plotly@5.4.0:")
+    depends_on("py-pyarrow")
+    depends_on("py-kaleido")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nanoplot/package.py]: Should all be type=build/run

File: var/spack/repos/builtin/packages/py-nanostat/package.py [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+    """Calculate statistics for Oxford Nanopore sequencing data and alignments"""
+
+    homepage = "https://github.com/wdecoster/nanostat"
+    pypi = "NanoStat/NanoStat-1.6.0.tar.gz"
+
+    maintainers("Pandapip1")
+
+    version("1.6.0", sha256="e45fa8d1ab49bdaed17596c26c0af148b44e4af46238391a8bb7a1b4cc940079")
+
+    depends_on("python@3")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nanostat/package.py]: Remove

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nanostat/package.py]: setuptools

File: var/spack/repos/builtin/packages/py-nanostat/package.py [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+    depends_on("py-nanoget@1.13.2:")
+    depends_on("py-nanomath@1.0.0:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nanostat/package.py]: build/run

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-kaleido/package.py (hunk 2 above)
Comment (Pandapip1) [reply var/spack/repos/builtin/packages/py-kaleido/package.py]: I'm not entirely sure how to set that up. Can you send me a documentation link?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-kaleido/package.py]: Something along the lines of:
```python
url = "https://github.com/plotly/Kaleido/archive/refs/tags/v0.2.1.post1.tar.gz"

version(...)  # update the checksum

depends_on("py-setuptools", type="build")

build_directory = join_path("repos", "kaleido", "py")
```
See https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html and https://spack.readthedocs.io/en/latest/packaging_guide.html

File: var/spack/repos/builtin/packages/py-nanoget/package.py [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+class PyNanoget(PythonPackage):
+    """Functions to extract information from Oxford Nanopore sequencing data and alignments"""
+
+    homepage = "https://github.com/wdecoster/nanoget"
+    pypi = "nanoget/nanoget-1.19.3.tar.gz"
+
+    maintainers("Pandapip1")
+
+    version("1.19.3", sha256="da981810edb1cbe42cbbfbe5fcf753f29bf5555204cd51256b28a284a036a71b")
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nanoget/package.py:18]: Missing build dep on setuptools. Also, if you sort the deps in the same order as in `setup.py`, it makes it easier for me to review.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by adamjstewart** [comment]
Also need to add a `py-python-deprecated` package

**Comment by Pandapip1** [comment]
> Also need to add a `py-python-deprecated` package

~~To which one?~~

EDIT: Oh, I need to add it as a package.

**Comment by Pandapip1** [comment]
@adamjstewart requesting re-review

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**